politeiad/backendv2/tstorebe/plugins/comments/cmds.go

```diff
@@ -1183,19 +1192,53 @@ func (p *commentsPlugin) cmdVotes(token []byte, payload string) (string, error)
 		return "", err
 	}
 
-	// Compile the comment vote digests for all votes that were cast
-	// by the specified user.
-	digests := make([][]byte, 0, 256)
-	for _, cidx := range ridx.Comments {
-		voteIdxs, ok := cidx.Votes[v.UserID]
-		if !ok {
-			// User has not cast any votes for this comment
-			continue
+	// Collect the requested page of comment vote digests
+	digests := make([][]byte, 0, p.votesPageSize)
+	var (
+		pageFirstIndex uint32 = (page - 1) * p.votesPageSize
+		pageLastIndex  uint32 = page * p.votesPageSize
+		idx            uint32 = 0
+	)
+	// Iterate over record index comments map deterministically; start from
+	// comment id 1 upwards.
+	for commentID := 1; commentID <= len(ridx.Comments); commentID++ {
+		// If digests page is full, then we are done
+		if len(digests) == int(p.votesPageSize) {
+			break
 		}
 
-		// User has cast votes on this comment
-		for _, vidx := range voteIdxs {
-			digests = append(digests, vidx.Digest)
+		cidx := ridx.Comments[uint32(commentID)]
+		switch {
+		// If no user ID filter is applied, we need to sort the Votes map keys,
```

Comment:
Comment could be improved.  I don't know how the comments are returned by just reading this.